### PR TITLE
New feature: Enlighten

### PR DESCRIPTION
**Pull request by Malabarba**
This PR adds support for the Enlighten feature in https://github.com/clojure-emacs/cider-nrepl/pull/296.

Take a file like the following:

screenshot-2016-02-05--053810

Turning on `cider-enlighten-mode` and evaluating the sexps one by one (`C-c C-k` doesn't work) will lead to this:

screenshot-2016-02-05--053747

### CODE REVIEW

**Comment by bbatsov** [comment]
You've just signed LT's death sentence. :-) The code looks good - all we need now is some documentation and a changelog entry. You can add this (or an animated gif) in the README.

**Comment by malchmih** [comment]
Simply awesome! :+1:

**Comment by arrdem** [comment]
Sweet!

**Comment by Malabarba** [comment]
Thanks. :-)

> all we need now is some documentation and a changelog entry.

Yeah, I'll start writing that now.

**Comment by bbatsov** [comment]
The docs look good. Guess you can squash the commits and we can merge this.

File: README.md [hunk 1]
```diff
@@ -808,6 +809,24 @@ will be used.
 (setq cider-stacktrace-fill-column 80)
 ```
 
+### Enlighten (Light Table)
```
Comment (bbatsov) [inline README.md]: Btw, maybe the reference to "Light Table" will be big misleading, as most people will think this is something like "InstaREPL".

**Comment by doubleagent** [comment]
This is awesome!

**Comment by Malabarba** [comment]
😊

**Pull request merged**